helix-core/src/main/java/org/apache/helix/messaging/handling/HelixTaskExecutor.java

```diff
@@ -209,7 +211,7 @@ public void registerMessageHandlerFactory(String type, MessageHandlerFactory fac
     if (prevItem == null) {
       ExecutorService newPool = Executors.newFixedThreadPool(threadpoolSize, new ThreadFactory() {
         @Override public Thread newThread(Runnable r) {
-          return new Thread(r, "HelixTaskExecutor-message_handle_thread");
+          return new Thread(r, "HelixTaskExecutor-message_handle_thread_" + thread_uid.getAndIncrement());
```

Comment:
Not related to your change. I'm wondering if we should make this daemon thread? @jiajunwang

Reply:
Daemon thread is just to make sure that JVM can shutdown automatically when main thread ( and all other non daemon thread) exit. It is similar to you Unix machine restart, daemon process won't block it from restarting, but your forground process, say intellij can block the restart of Unix machine.